cmd/query/app/server_test.go

```diff
@@ -43,13 +47,40 @@ func TestServer(t *testing.T) {
 	flagsSvc := flags.NewService(ports.AgentAdminHTTP)
 	flagsSvc.Logger = zap.NewNop()
 
-	querySvc := &querysvc.QueryService{}
+	spanReader := &spanstoremocks.Reader{}
+	dependencyReader := &depsmocks.Reader{}
+
+	querySvc := querysvc.NewQueryService(spanReader, dependencyReader, querysvc.QueryServiceOptions{})
+
 	tracer := opentracing.NoopTracer{}
 
-	server := NewServer(flagsSvc, querySvc, &QueryOptions{Port: ports.QueryAdminHTTP,
-		BearerTokenPropagation: true}, tracer)
+	server := NewServer(flagsSvc, querySvc,
+		&QueryOptions{Port: ports.QueryAdminHTTP, BearerTokenPropagation: true},
```

Comment:
Changing the listening port here to -

```
		&QueryOptions{Port: ports.QueryHTTP, BearerTokenPropagation: true},
```

.. and using `cmuxServer.MatchWithWriters(...)` as suggested by @32leaves seems to work for me. 

We definitely needed an integration test with cmux though, thanks for pointing out!